api/_utils.ts

```diff
@@ -1992,3 +1993,89 @@ export function paramToArray<T extends undefined | string | string[]>(
   if (!param) return;
   return Array.isArray(param) ? param : [param];
 }
+
+export type TokenOptions = {
+  chainId: number;
+  address: string;
+};
+
+const TTL_TOKEN_INFO = 30 * 24 * 60 * 60; // 30 days
+
+function tokenInfoCache(params: TokenOptions) {
+  return makeCacheGetterAndSetter(
+    buildInternalCacheKey("tokenInfo", ...Object.values(params)),
```

Comment:
NIT: should we make this positional and explicit args? like
```suggestion
    buildInternalCacheKey("tokenInfo", params.address, params.chainId),
```
This would prevent duplicate records in the cache for 
```js
getCachedTokenInfo({ chainId, address })
// below would create separate cache key with current implementation
getCachedTokenInfo({ address, chainId })
```